pkg/leader/leader.go

```diff
@@ -109,7 +109,33 @@ func Become(ctx context.Context, lockName string) error {
 			log.Info("Became the leader.")
 			return nil
 		case apierrors.IsAlreadyExists(err):
-			log.Info("Not the leader. Waiting.")
+			existing := &corev1.ConfigMap{}
+			key := crclient.ObjectKey{Namespace: ns, Name: lockName}
+			err := client.Get(ctx, key, existing)
+			if err != nil {
+				return err
+			}
+			leaderPod := &corev1.Pod{}
+			key = crclient.ObjectKey{Namespace: ns, Name: existing.ObjectMeta.OwnerReferences[0].Name}
+			err = client.Get(ctx, key, leaderPod)
+			if err != nil {
+				return err
+			}
+
+			podFailed := leaderPod.Status.Phase == corev1.PodFailed
+			podEvicted := leaderPod.Status.Reason == "Evicted"
+			if podFailed && podEvicted {
+				log.Info("Operator pod with leader lock has been evicted.", "Leader pod", leaderPod.Name)
+				// TODO Should this behavior be configurable?
```

Comment:
I want to highlight this question for reviewers.

Reply:
IHMO: We can move forward with it as default behaviour, and then, if the future the needs bring up we can improve and make it configurable.

Reply:
I've removed the comment, but will leave this conversation unresolved for other reviewers to comment.

Reply:
Agree with @camilamacedo86. Let's leave it as is for now. If a use case comes up where someone wants to have an evicted leader pod deadlock situation that must be manually resolved, we can add that in later.